src/ZiraLink.Api.Application/Services/CustomerService.cs

```diff
@@ -126,23 +116,18 @@ public async Task ChangePasswordAsync(string userId, string currentPassword, str
                 throw new ArgumentNullException(nameof(newPassword));
 
             var customer = await _dbContext.Customers.AsNoTracking().SingleOrDefaultAsync(x => x.ExternalId == userId, cancellationToken);
-            if (customer == null) throw new NotFoundException(nameof(Customer));
-
-            var client = await InitializeHttpClientAsync(cancellationToken);
-
+            if (customer == null)
+                throw new NotFoundException(nameof(Customer), new List<KeyValuePair<string, object>>() { new KeyValuePair<string, object>(nameof(Customer.ExternalId), userId) });
+             
             var jsonObject = new
             {
                 UserId = userId,
                 CurrentPassword = currentPassword,
                 NewPassword = newPassword
             };
-            var content = new StringContent(JsonSerializer.Serialize(jsonObject), Encoding.UTF8, "application/json");
-            var baseUri = _idsUri;
-            var uri = new Uri(baseUri, "User/ChangePassword");
-            var response = await client.PatchAsync(uri.ToString(), content);
 
-            var responseString = await response.Content.ReadAsStringAsync(cancellationToken);
-            var userCreationResult = JsonSerializer.Deserialize<ApiResponse<string>>(responseString, new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
+            var userCreationResult = await _identityService.ChangePasswordAsync(jsonObject, cancellationToken);
+         
             if (!userCreationResult.Status)
                 throw new ApplicationException("User creation on identity server failed");
```

Comment:
Error message should be something like `Changing password failed`

Reply:
Done.